src/braket/analog_hamiltonian_simulator/rydberg/validators/program.py

```diff
@@ -34,7 +39,64 @@ def local_detuning_pattern_has_the_same_length_as_atom_array_sites(cls, values):
             pattern_size = len(local_detuning["magnitude"]["pattern"])
             if num_sites != pattern_size:
                 raise ValueError(
-                    f"The length of pattern ({pattern_size}) of shifting field {idx} must equal "
+                    f"The length of pattern ({pattern_size}) of local detuning {idx} must equal "
                     f"the number of atom array sites ({num_sites})."
                 )
         return values
+
+    # If there is local detuning, the net value of detuning for each atom
+    # should not exceed a max detuning value
+    @root_validator(pre=True, skip_on_failure=True)
+    def net_detuning_must_not_exceed_max_net_detuning(cls, values):
+        capabilities = values["capabilities"]  # device_capabilities
+
+        # Extract the program and the fields
+        program = deepcopy(values)
+        del program["capabilities"]
+        program = Program.parse_obj(program)
+        driving_fields = program.hamiltonian.drivingFields
+        local_detuning = program.hamiltonian.localDetuning
+
+        # If no local detuning, return
+        if not len(local_detuning):
+            return values
+
+        detuning_times = [
+            local_detune.magnitude.time_series.times for local_detune in local_detuning
+        ]
+
+        # Merge the time points for different shifting terms and detuning term
+        time_points = sorted(list(set(sum(detuning_times, []))))
+        for driving_field in driving_fields:
+            time_points = sorted(list(set(time_points + driving_field.detuning.time_series.times)))
+
+        # Get the time-dependent functions for the detuning and shifts
+        _, detuning_coefs, shift_coefs = _get_coefs(program, time_points)
+
+        # Get the detuning pattern
+        detuning_patterns = [local_detune.magnitude.pattern for local_detune in local_detuning]
+
+        # For each time point, check that each atom has net detuning less than the threshold
+        for time_ind, time in enumerate(time_points):
+            for atom_index in range(len(detuning_patterns[0])):
+                # Get the contributions from global detuning at the time point
+                detuning_to_check = 0
+                for detuning_coef in detuning_coefs:
+                    detuning_to_check += detuning_coef[time_ind]
+
+                # Get the contributions from local detuning at the time point
+                for detuning_pattern, shift_coef in zip(detuning_patterns, shift_coefs):
+                    detuning_to_check += shift_coef[time_ind] * float(detuning_pattern[atom_index])
```

Comment:
The method name says its checking if the net values doesn't exceed the max detuning value, but it retrieving the contributions, I need a bit more context on this to understand why this is part of this method.